src/Microsoft.DotNet.Darc/src/Darc/Helpers/UxHelpers.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.Maestro.Client.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Darc
+{
+    public static class UxHelpers
+    {
+        /// <summary>
+        ///     Resolve a channel substring to an exact channel, or print out potential names if more than one, or none, match.
+        /// </summary>
+        /// <param name="remote">Remote for retrieving channels</param>
+        /// <param name="desiredChannel">Desired channel</param>
+        /// <returns>Channel, or null if no channel was matched.</returns>
+        public static async Task<Channel> ResolveSingleChannel(IRemote remote, string desiredChannel)
+        {
+            // Retrieve the channel by name, matching substring. If more than one channel 
+            // matches, then let the user know they need to be more specific
+            IEnumerable<Channel> channels = (await remote.GetChannelsAsync());
+            List<Channel> matchingChannels = channels.Where(c => c.Name.Contains(desiredChannel, StringComparison.OrdinalIgnoreCase)).ToList();
```

Comment:
>.ToList() [](start = 133, length = 9)

You can avoid this ToList since all the functions on matchingChannels are in IEnumerable

Reply:
Count is not...

---
In reply to: [277416602](https://github.com/dotnet/arcade-services/pull/362#discussion_r277416602) [](ancestors = 277416602)